app/models/base_organisation.rb

```diff
@@ -9,16 +9,21 @@ class BaseOrganisation < ApplicationRecord
     if: proc{|org| org.donation_info.present?}
   validates :name, presence: { message: "Name can't be blank"}
   validates :description, presence: { message: "Description can't be blank"}
-  has_many :category_organisations, :foreign_key => :organisation_id
-  has_many :categories, :through => :category_organisations, :foreign_key => :organisation_id
+  has_many :category_organisations, foreign_key: :organisation_id
+  has_many :categories, through: :category_organisations, foreign_key: :organisation_id
+
   # For the geocoder gem
   geocoded_by :full_address
   after_validation :geocode, if: -> { run_geocode? }
-  self.table_name = "organisations"
+  self.table_name = 'organisations'
 
   def run_geocode?
-    ## http://api.rubyonrails.org/classes/ActiveModel/Dirty.html
-    address_changed? or postcode_changed? or (address.present? and not_geocoded?)
+    # trigger geocode if I have an address but no coordinates and
+    # the object is being created
+    return (!address.empty? and not_geocoded?) if id.nil?
```

Comment:
hi there @tansaku, I changed the logic to avoid API calls when we are creating an organisation, and that organisation already has latitude, longitude and an address.

Since I'm checking that the address should not be empty, are we preventing the user to enter `""` (empty string) as an address right? One thing that we should consider is that we have as default value the empty string:

```
  create_table "organisations", id: :serial, force: :cascade do |t|
    t.string "name", default: "", null: false
    t.string "address", default: "", null: false
    t.string "postcode", default: "", null: false
    t.string "email", default: "", null: false
    t.text "description", default: "", null: false
    t.string "website", default: "", null: false
    t.string "telephone", default: "", null: false
    t.datetime "created_at"
    t.datetime "updated_at"
    t.float "latitude"
    t.float "longitude"
    t.boolean "gmaps"
    t.text "donation_info", default: "", null: false
    t.boolean "publish_address", default: false
    t.boolean "publish_phone", default: false
    t.boolean "publish_email", default: true
    t.datetime "deleted_at"
    t.string "type", default: "Organisation"
    t.boolean "non_profit"
    t.string "slug"
    t.index ["slug"], name: "index_organisations_on_slug", unique: true
  end
```

Should we leave it like that, or should we add a migration to remove that default value?

Reply:
I think it should be okay to enter the empty string, we just want to make sure that we don't geocode when that's the case